src/hotspot/share/ci/ciFlags.cpp

```diff
@@ -92,5 +92,5 @@ void ciFlags::print_member_flags(outputStream* st) {
 // ------------------------------------------------------------------
 // ciFlags::print
 void ciFlags::print(outputStream* st) {
-  st->print(" flags=%x", _flags);
+  st->print(" flags=%x", _flags.as_unsigned_short());
```

Comment:
Here, and elsewhere, are we relying on an implicit widening of the u2 result to int so that the format specifier is correct?

Reply:
Yes, we are.

Reply:
IMO you could just call `as_int()` here. All other usages of `ciFlags::as_int()` are in printing code. 

Ideally, `ciField::print()` could use `ciFlags::print()`, but such cleanup can be done separately.